src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -155,24 +159,24 @@ public sealed class StructField
             string name,
             DataType dataType,
             bool isNullable = true,
-            JObject metadata = null)
+            JsonElement? metadata = null)
```

Comment:
Why is a nullable `JsonElement` being used? `JsonElement` is a struct. I would use `JsonElement metadata = default` instead.